docs/en/03_Upgrading/index.md

```diff
@@ -37,123 +89,1015 @@ and also checking the `composer.json` files in [recipe-core](https://github.com/
+
+Like the `upgrade` command, `inspect` gets its list of API changes from `.upgrade.yml` files. So you may get upgrade suggestions and substitution from third party modules. You can even include your own project specific changes in your `.upgrade.yml` if you want.
+
+#### Sample output of the `inspect` command
+Here's some sample output of what you might get back the `inspect` command.
+
+```bash
+/var/www/SS_acme > upgrade-code inspect ./mysite/Models/Coyote.php
+Running post-upgrade on "/var/www/SS_acme/mysite/code/Models/Coyote.php"
+[2018-06-06 13:35:38] Applying ApiChangeWarningsRule to Coyote.php...
+modified:	Coyote.php
+@@ -68,7 +68,7 @@
+     {
+         // Getting a reference to Coyote's list of crazy ideas
+-        $manyManyRelation = $this->manyManyComponent('CrazyIdeas');
++        $manyManyRelation = $this->getSchema()->manyManyComponent('CrazyIdeas');
+         return $manyManyRelation;
+     }
+
+Warnings for Coyote.php:
+ - Coyote.php:20 SS_Cache: Using symfony/cache now (https://docs.silverstripe.org/en/4/changelogs/4.0.0#cache)
+ - Coyote.php:42 SilverStripe\Control\Director::setUrlParams(): Method removed
+ - Coyote.php:71 SilverStripe\ORM\DataObject->manyManyComponent(): DataObject->manyManyComponent() moved to DataObjectSchema. Access through getSchema(). You must manually add static::class as the first argument to manyManyComponent()
+Changes not saved; Run with --write to commit to disk
+```
+
+### Manually update deprecated API references
+
+SilverStripe 4 introduces thousands of small and big API changes. To update deprecated API references manually, you have to go through each one of your project files.
+
+[Read the SilverStripe 4 change logs](/changelogs/4.0.0/) for a comprehensive list of what has changed.
+
+### Finalising the deprecated API update
+At this stage, your site should be using only SilverStripe 4 API logic.
+
+You still have some minor clean up tasks and configuration tweaks to apply, but you're almost done.
+
+This is a good point to commit your changes to your source control system before moving on to the next step.
+
+## Step 6 - Update your entry point {#step6}
+The location of SilverStripe's _entry file_ has changed. Your project and server environment will need
+to adjust the path to this file from `framework/main.php` to `index.php`.
+
+### Update your entry point
+You can get a copy of the SilverStripe 4 `index.php` file at:
+* `vendor/silverstripe/recipe-core/public/index.php` if you are upgrading to SilverStripe 4.1 or above
+* `vendor/silverstripe/recipe-core/index.php` if you are upgrading to SilverStripe 4.0
+
+If you've modified your SilverStripe 3 `index.php`, you'll need to reconcile those changes with the index.php file you got from `recipe-core`. Otherwise, just use the generic `index.php` file `recipe-core` provides.
```

Comment:
It's not really possible to modify your `index.php` file in SS 3 because the default "entry point" is `framework/main.php`... If someone has reworked their entry point to be an `index.php` file I think they are probably advanced enough to know about it and that this does not need mentioning. It also directly contradicts what you've said above about the old entry point and so could be confusing for readers

Reply:
The main reason why I mention the entry point has changed is so people who have hadcoded their rewrite rule into their VHOST config rather than rely on `.htaccess` no what they need to update.

While I agree it would be very unusual for someone to update their `index.php` file in SS3, it's a scenario that the webroot command handles, so it seems right to mention it in the manual process.

I've rename the header to _Update your index.php file_ to avoid implying that `index.php` was the entry point in SS3.

Reply:
>I've renamed the header to Update your index.php file to avoid implying that index.php was the entry point in SS3.

But it wasn't... I'm worried we'll confuse 95% of readers just to cater to 5% that have custom routing of requests...